packages/next/src/lib/typescript/runTypeCheck.ts

```diff
@@ -38,11 +37,9 @@ export async function runTypeCheck(
       incremental: false,
     }
   }
-  const requiredConfig = getRequiredConfiguration(ts)
 
   const options = {
     ...effectiveConfiguration.options,
-    ...requiredConfig,
```

Comment:
I think the user's config should be respected (e.g. `nodenext`) but Next.js also has to provide a fallback. So I recommend swapping their order here: 

```ts
  const requiredConfig = getRequiredConfiguration(ts)

  const options = {
    ...requiredConfig,
    ...effectiveConfiguration.options,
```

Reply:
That makes sense, I can update.